exercises/luhn/example.exs

```diff
@@ -1,59 +1,44 @@
 defmodule Luhn do
-  @doc """
-  Calculates the total checksum of a number
-  """
-  @spec checksum(String.t()) :: integer
-  def checksum(number) do
-    number
-    |> String.reverse
-    |> String.graphemes
-    |> Enum.with_index
-    |> Enum.map(&do_checksum/1)
-    |> Enum.sum
-  end
-
-  defp do_checksum({digit, index}) do
-    digit
-    |> String.to_integer
-    |> double_if_even_index(index)
-    |> compact
-  end
-
-  defp compact(num) when num < 10, do: num
-  defp compact(num) do
-    num
-    |> Integer.digits
-    |> Enum.sum
-    |> compact
-  end
 
   @doc """
   Checks if the given number is valid via the luhn formula
   """
   @spec valid?(String.t()) :: boolean
   def valid?(number) do
-    number |> checksum |> rem(10) == 0
-  end
 
-  @doc """
-  Creates a valid number by adding the correct
-  checksum digit to the end of the number
-  """
-  @spec create(String.t()) :: String.t()
-  def create(number) do
-    number <> do_create(number)
+    with  number       <- String.replace(number, " ", ""),
```

Comment:
So while this is technically correct, it's probably not the best usage of `with`. It's sort of a combination of the kind of railroad programming that it's designed for and some boolean logic. Plus the variable re-assignment is slightly confusing to me. On line 10, 11 and 12 you might not really know which `number` you're talking about. It's not really a blocker for deploying this, but if it were up to me, I'd rewrite this along the lines of the following:

```elixir
def valid?(number) do
  without_spaces = String.replace(number, " ", "")
  case Integer.parse(without_spaces) do
    {_, ""} ->
      String.length(without_spaces) > 1 && checksum(without_spaces)
    _ -> false
end
```